proxy/director.go

```diff
@@ -65,6 +66,13 @@ func (d *director) refresh() {
 		}
 		endpoints = append(endpoints, newEndpoint(*uu))
 	}
+
+	// shuffle array to avoid connections being "stuck" to a single endpoint
+	for i := range endpoints {
```

Comment:
Correct. This is a quick fix for the short-term.  When we coalesce watches, we will need the watches to be "sticky" to a given endpoint - or at least all watches for a given key.